packages/@lwc/engine/src/framework/wiring.ts

```diff
@@ -155,7 +155,27 @@ function createConnector(vm: VM, name: string, wireDef: WireDef): WireAdapter {
         // This wire has dynamic parameters: we wait for the component instance is created and its values set
         // in order to call the update(config) method.
         Promise.resolve().then(() => {
-            computeConfigAndUpdate = createConfigWatcher(vm, wireDef, updateConnectorConfig);
+            // there is a special case: when all the config params are undefined, the config on the
```

Comment:
After speaking to @jodarove over the phone, we think that this code really doesn't belong here, it belongs to the register service for legacy adapters only. Setting values to undefined is just fine, and seems like we do have legacy adapters that are relying on the existing bug to behave based on undefined value. Moving this code there means that new adapters, and open source adapters will not see this buggy behavior.